### PR TITLE
Setup IRSA module

### PR DESCRIPTION
**Pull request by lucasrod16**
Relates to https://github.com/defenseunicorns/uds-package-dubbd/issues/337

### CODE REVIEW

File: examples/complete/main.tf [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+    {
+      Version = "2012-10-17",
+      Statement = [
+        {
+          Effect = "Allow",
+          Action = [
+            "route53:ChangeResourceRecordSets"
+          ]
+          Resource = [
+            "arn:aws:route53:::hostedzone/*"
```
Comment (justin-o12) [inline examples/complete/main.tf:67]:
```suggestion
data "aws_partition" "current" {}

data "aws_region" "current" {}

resource "random_id" "unique_id" {
  byte_length = 4
}

resource "aws_iam_policy" "loki_policy" {
  name        = "LokiPolicy-${random_id.unique_id.hex}"
  path        = "/"
  description = "Policy for S3 access."

  policy = jsonencode({
    Version = "2012-10-17"
    Statement = [
      {
        Effect   = "Allow"
        Action   = ["s3:ListBucket"]
        Resource = ["arn:${data.aws_partition.current.partition}:s3:::dummy-bucket"]
      },
      {
        Effect   = "Allow"
        Action   = ["s3:*Object"]
        Resource = ["arn:${data.aws_partition.current.partition}:s3:::dummy-bucket/*"]
      },
      {
        Effect = "Allow"
        Action = [
          "kms:GenerateDataKey",
          "kms:Decrypt"
        ]
        Resource = ["arn:${data.aws_partition.current.partition}:kms:${data.aws_region.current.name}:${data.aws_caller_identity.current.account_id}:key/dummy-key"]
      }
    ]
  })
}

resource "aws_iam_policy" "external_dns_policy" {
  name        = "AllowExternalDNSUpdates-${random_id.unique_id.hex}"
  path        = "/"
  description = "Policy for external-dns to create records in route53 hosted zones."

  # Terraform expression result to valid JSON syntax.
  policy = jsonencode(
    {
      Version = "2012-10-17",
      Statement = [
        {
          Effect = "Allow",
          Action = [
            "route53:ChangeResourceRecordSets"
          ]
          Resource = [
            "arn:${data.aws_partition.current.partition}:route53:::hostedzone/*"
```

Comment (justin-o12) [reply examples/complete/main.tf:67]: This is mostly just trying to parameterize the region, partition, and account ID for the ARNs.

Comment (lucasrod16) [reply examples/complete/main.tf]: @justin-o12 changes applied! I also parameterized the ARNs for the velero policy in the example module

**Review by justin-o12 — APPROVED** [review]
:rocket:

**Pull request merged**